frontend/public/components/droppable-edit-yaml.tsx

```diff
@@ -20,6 +20,7 @@ const boxTarget = {
     }
   },
 };
+let fileUploadContents = '';
```

Comment:
Don't track state in a top-level variable. This will be remember if you leave the page and come back, and it prevents you from ever being able to use the component twice on the same page. Better to use `useRef` in the component.

Reply:
@spadgett this is a class based component, would you suggest converting to an FC or is there a way to accomplish the same pre - hooks?

Reply:
@spadgett I'll check out React.createRef() for class components. thanks!

Reply:
@spadgett I pulled the top level variable into a private class variable. I think this is doing what you'd want with the useRef, but in a class environment.

Reply:
Yes, sorry. Class variable is right for class-based components.